File: cloud-ui/src/api/order/order.ts
import createRequest from '@/api/request'
import { SERVICE_PREFIX } from '@/config/service.config'

const orderApi = createRequest(SERVICE_PREFIX.ORDER)

export interface OrderItem {
  id: number
  productId: number
  skuId: number
  productName: string
  skuName: string
  productImage: string
  quantity: number
  price: number
  totalAmount: number
  itemStatus: number
}

export interface OrderMain {
  id: number
  orderNo: string
  userId: number
  totalAmount: number
  actualPayAmount: number
  shippingFee: number
  discountAmount: number
  orderStatus: number
  paymentStatus: number
  paymentType: number
  paymentTime: string
  consigneeName: string
  consigneePhone: string
  consigneeAddress: string
  remark: string
  createTime: string
  items: OrderItem[]
}

export interface CreateOrderRequest {
  cartItemIds: number[]
  shippingAddressId: number
  couponId?: number
  remark?: string
}

export interface OrderListResponse {
  records: OrderMain[]
  total: number
  size: number
  current: number
}

/**
 * 创建订单
 */
export const createOrder = (data: CreateOrderRequest) => {
  return orderApi.post<{ orderId: number; orderNo: string }>('/order/create', data)
}

/**
 * 获取订单列表
 */
export const getOrderList = (params: {
  current: number
  size: number
  orderStatus?: number
}) => {
  return orderApi.get<OrderListResponse>('/order/list', { params })
}

/**
 * 获取订单详情
 */
export const getOrderDetail = (orderId: number) => {
  return orderApi.get<OrderMain>(`/order/detail/${orderId}`)
}

/**
 * 取消订单
 */
export const cancelOrder = (orderId: number) => {
  return orderApi.put(`/order/cancel/${orderId}`)
}

/**
 * 确认收货
 */
export const confirmOrder = (orderId: number) => {
  return orderApi.put(`/order/confirm/${orderId}`)
}

/**
 * 支付订单
 */
export const payOrder = (orderId: number, paymentType: number) => {
  return orderApi.post<{ payUrl: string }>(`/order/pay/${orderId}`, { paymentType })
}

/**
 * 删除订单
 */
export const deleteOrder = (orderId: number) => {
  return orderApi.delete(`/order/delete/${orderId}`)
}

File: cloud-order/src/main/java/com/cloud/order/entity/Order.java
package com.cloud.order.entity;

import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单主实体
 * 用于表示订单的基本信息
 */
@Data
public class Order implements Serializable {
    private static final long serialVersionUID = 1L;

    /**
     * 订单ID
     */
    private Long id;

    /**
     * 用户ID
     */
    private Long userId;

    /**
     * 订单总金额
     */
    private BigDecimal totalAmount;

    /**
     * 实际支付金额
     */
    private BigDecimal actualPayAmount;

    /**
     * 运费
     */
    private BigDecimal shippingFee;

    /**
     * 优惠券ID
     */
    private String couponId;

    /**
     * 优惠金额
     */
    private BigDecimal discountAmount;

    /**
     * 订单状态: 1(待支付), 2(待发货), 3(已发货), 4(已完成), 5(已取消), 6(已关闭)
     */
    private Integer orderStatus;

    /**
     * 支付状态: 1(未支付), 2(已支付), 3(已退款), 4(部分退款)
     */
    private Integer paymentStatus;

    /**
     * 支付方式: 1(支付宝), 2(微信支付), 3(信用卡)
     */
    private Integer paymentType;

    /**
     * 支付时间
     */
    private LocalDateTime paymentTime;

    /**
     * 收货地址ID
     */
    private String shippingAddressId;

    /**
     * 收货人姓名
     */
    private String consigneeName;

    /**
     * 收货人电话
     */
    private String consigneePhone;

    /**
     * 收货人详细地址
     */
    private String consigneeAddress;

    /**
     * 订单备注
     */
    private String remark;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

    /**
     * 删除标记: 0(未删除), 1(已删除)
     */
    private Integer deleted;
}